dot/network/service.go

```diff
@@ -410,3 +410,8 @@ func (s *Service) Peers() []common.PeerInfo {
 	}
 	return peers
 }
+
+// NodeRoles Returns the roles the node is running as.
```

Comment:
I think this should be part of the core service rather than the network service? maybe? we pass this configuration value down from the dot configuration to the core and network services and the network service is less the source of truth for this sort of thing than core but probably ok as is - kinda just thinking this out loud.

Reply:
I'd like to leave this as is for now.  I agree it makes sense to include this in core, however the RPC module that's using this doesn't have reference to coreAPI, it currently has access to networkAPI and systemAPI.